libs/math/tests/math/all_tests/fixed_point/fixed_point.cpp

```diff
@@ -786,39 +807,172 @@ TEST(FixedPointTest, Exponential_32_32)
 }
 
-TEST(FixedPointTest, Pow_32_32)
+TEST(FixedPointTest, Pow_16_16_negative_x)
 {
-  fetch::fixed_point::FixedPoint<32, 32> x{-1.6519711627625};
+  fp32   step{0.01};
+  fp32   x{-10};
+  fp32   y{-4};
+  double max_error = 0, avg_error = 0;
+  size_t iterations = 0;
+  double tolerance  = (double)fp32::TOLERANCE;
+  for (; x < 10.0; x += step)
+  {
+    for (; y < 4; ++y)
+    {
+      fp32   e     = fp32::Pow(x, y);
+      double r     = std::pow((double)x, (double)y);
+      double delta = std::abs((double)e - r);
+      max_error    = std::max(max_error, delta);
+      avg_error += delta;
+      iterations++;
+    }
+  }
+  avg_error /= (double)iterations;
+  EXPECT_NEAR(max_error, 0.0, tolerance);
+  EXPECT_NEAR(avg_error, 0.0, tolerance);
+  std::cout << "Pow: max error = " << max_error << std::endl;
+  std::cout << "Pow: avg error = " << avg_error << std::endl;
+}
+
+TEST(FixedPointTest, Pow_32_32_positive_x)
+{
+  fetch::fixed_point::FixedPoint<32, 32> a{-1.6519711627625};
   fetch::fixed_point::FixedPoint<32, 32> two{2};
   fetch::fixed_point::FixedPoint<32, 32> three{3};
-  fetch::fixed_point::FixedPoint<32, 32> y{1.8464393615723};
-  fetch::fixed_point::FixedPoint<32, 32> e1 = fp64::Pow(x, two);
-  fetch::fixed_point::FixedPoint<32, 32> e2 = fp64::Pow(x, three);
-  fetch::fixed_point::FixedPoint<32, 32> e3 = fp64::Pow(two, y);
+  fetch::fixed_point::FixedPoint<32, 32> b{1.8464393615723};
+  fetch::fixed_point::FixedPoint<32, 32> e1 = fp64::Pow(a, two);
+  fetch::fixed_point::FixedPoint<32, 32> e2 = fp64::Pow(a, three);
+  fetch::fixed_point::FixedPoint<32, 32> e3 = fp64::Pow(two, b);
 
   EXPECT_NEAR((double)e1 / std::pow(-1.6519711627625, 2), 1.0, (double)fp64::TOLERANCE);
   EXPECT_NEAR((double)e2 / std::pow(-1.6519711627625, 3), 1.0, (double)fp64::TOLERANCE);
-  // This test has ~1.4e-6 error
-  EXPECT_NEAR((double)e3 / std::pow(2, 1.8464393615723), 1.0, 1.5e-6);
-  EXPECT_THROW(fp64::Pow(fp64::CONST_ZERO, fp64::CONST_ZERO), std::runtime_error);
-  EXPECT_THROW(fp64::Pow(x, x), std::runtime_error);
+  EXPECT_NEAR((double)e3 / std::pow(2, 1.8464393615723), 1.0, (double)fp64::TOLERANCE);
+  EXPECT_TRUE(fp64::isNaN(fp64::Pow(fp64::_0, fp64::_0)));
+  EXPECT_TRUE(fp64::isNaN(fp64::Pow(a, a)));
+
+  fp64   step{0.0001};
+  fp64   x{0.0001};
+  fp64   y{0.001};
+  double max_error = 0, avg_error = 0;
+  size_t iterations = 0;
+  double tolerance  = (double)fp64::TOLERANCE;
+  for (; x < 100.0; x += step)
+  {
+    for (; y < 40.5; y += step)
+    {
+      fp64   e     = fp64::Pow(x, y);
+      double r     = std::pow((double)x, (double)y);
+      double delta = std::abs((double)e - r);
+      max_error    = std::max(max_error, delta);
+      avg_error += delta;
+      iterations++;
+    }
+  }
+  avg_error /= (double)iterations;
+  EXPECT_NEAR(max_error, 0.0, tolerance);
+  EXPECT_NEAR(avg_error, 0.0, tolerance);
+  std::cout << "Pow: max error = " << max_error << std::endl;
```

Comment:
this was mostly for my own information and to put in the comments of the mathematical functions, so that people have an idea of the expected accuracy, but sure I can comment out or remove from the tests altogether.

Reply:
I will comment out these prints for normal test runs.